### DISCUSSION TITLE
When you modify the name of the profile, log out, it will reset to the old name!

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#4978] [Plugins]
![화면 캡처 2024-07-18 144809](https://github.com/user-attachments/assets/46d4d5f0-b7cc-446b-a19d-c6f2c1f9bd7a)

Hello!

If you modify the name in the kimai profile, log out, it will be initialized to the previous name.

How can I fix this error?'

Please check.

**Comment by seleondevel** [comment]
@kevinpapst Haven't you noticed any problems? Which part of the code should I check?

**Comment by kevinpapst** [comment]
Always verify your findings in the official demo before posting them.
This is likely connected to some change you made.

**Comment by seleondevel** [comment]
@kevinpapst As long as it's a regular account. However, if you rename, log out, and log in again to your Google SAML account, the profile name will reset to the previous one.

There is no part to reset the profile name in the kimai/config/packages/local.yaml below, where can I find it?

https://www.kimai.org/documentation/saml-google.html


saml:
        provider: google
        activate: true
        title: Login with SAML
        mapping:
            - { saml: $Email, kimai: email }
            - { saml: $FirstName $LastName, kimai: alias }
        roles:
            resetOnLogin: false
            attribute: Groups
            mapping:
                - { saml: Admin, kimai: ROLE_SUPER_ADMIN }
                - { saml: Manager, kimai: ROLE_ADMIN }
                - { saml: Teamlead, kimai: ROLE_TEAMLEAD }

**Comment by seleondevel** [comment]
@kevinpapst  

I checked and found that $this->hydrateUser(User, $token) in the file below. This part changes the profile name.
kimai/src/SamlSamlProvider.php (Line 52)

We use FirstName and LastName as one name without distinction, how should we handle this?

**Reply by kevinpapst** [reply]
It is in your local.yaml config then. Simply remove the mapping.

**Reply by seleondevel** [reply]
@kevinpapst Oh! Thanks to you, I solved it well! Thank you!

**Comment by Solution1997** [comment]
> ![화면 캡처 2024-07-18 144809](https://private-user-images.githubusercontent.com/165639563/349828346-46d4d5f0-b7cc-446b-a19d-c6f2c1f9bd7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MjY1NjEsIm5iZiI6MTcyNTYyNjI2MSwicGF0aCI6Ii8xNjU2Mzk1NjMvMzQ5ODI4MzQ2LTQ2ZDRkNWYwLWI3Y2MtNDQ2Yi1hMTlkLWM2ZjJjMWY5YmQ3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNlQxMjM3NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDc2OThjYTE4YjA5YTQ3NGZkNjI4ZTYwMzgyMTliMTU2MjcyMmE0NzU0OWQxZmIyZDk1ODQzMTU0ZTIzMTYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NUapKXaDmvKpixnqOZaGLFuje2jC-AyGIgU2qTwI1pc)
> 
> Hello!
> 
> If you modify the name in the kimai profile, log out, it will be initialized to the previous name.
> 
> How can I fix this error?'
> 
> Please check.

Please can you also guide me on admin php login link password reset...?
